crates/runtime/src/catalogconnector/glue/provider.rs

```diff
@@ -0,0 +1,475 @@
+        )
+        .await
+        .map_err(|message| super::Error::ConfigurationLoadingFailed { message })?;
+        let glue = Client::new(&config);
+
+        let get_databases_output = glue
+            .get_databases()
+            .send()
+            .await
+            .context(super::GetDatabasesSnafu)?;
+
+        let mut databases = HashMap::new();
+        for db in get_databases_output.database_list {
+            if !database_might_match(&db.name, &catalog.orig_include) {
+                tracing::debug!("skipping database {}", &db.name);
+                continue;
+            }
+
+            let get_tables_output = glue
+                .get_tables()
+                .database_name(&db.name)
+                .send()
+                .await
+                .context(super::GetTablesSnafu)?;
+
+            let table_names = get_tables_output
+                .table_list
+                .unwrap_or_default()
+                .into_iter()
+                .filter(|t| {
+                    !matches!(TableType::from(t), TableType::Unsupported)
+                        && is_included(catalog.include.as_ref(), &db.name, t.name())
+                })
+                .collect::<Vec<_>>();
+
+            if !table_names.is_empty() {
+                databases.insert(db.name, table_names);
+            }
+        }
+
+        let inner = Arc::new(GlueCatalogState::new(
+            databases, config, parameters, runtime,
+        ));
+
+        Ok(Self { inner })
+    }
+}
+
+impl CatalogProvider for GlueCatalogProvider {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema_names(&self) -> Vec<String> {
+        self.inner.databases.keys().cloned().collect()
+    }
+
+    fn schema(&self, name: &str) -> Option<Arc<dyn datafusion::catalog::SchemaProvider>> {
+        if self.inner.databases.contains_key(name) {
+            let schema_provider = GlueSchemaProvider {
+                database: name.to_string(),
+                state: Arc::clone(&self.inner),
+            };
+            Some(Arc::new(schema_provider))
+        } else {
+            None
+        }
+    }
+}
+
+#[async_trait]
+impl RefreshableCatalogProvider for GlueCatalogProvider {
+    async fn refresh(&self) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        // TODO: #6012
+        Ok(())
+    }
+}
+
+impl GlueSchemaProvider {
+    async fn create_iceberg_provider(
+        &self,
+        name: &str,
+        table: &Table,
+    ) -> DFResult<Option<Arc<dyn TableProvider>>> {
+        let mut props = Vec::new();
+        if let Some(provider) = self.state.config.credentials_provider() {
+            let creds = provider
+                .provide_credentials()
+                .await
+                .map_err(|e| DataFusionError::External(Box::new(e)))?;
+            props.push((S3_ACCESS_KEY_ID, creds.access_key_id().to_string()));
+            props.push((S3_SECRET_ACCESS_KEY, creds.secret_access_key().to_string()));
```

Comment:
If the creds are temporary, then this will stop working once they expire.

Reply:
Ideally we let the underlying provider have the same inputs as the glue connector itself, i.e. only if an explicit key_id/secret_access_key is provided then we pass that